policy-test/src/curl.rs

```diff
@@ -0,0 +1,233 @@
+            k8s::api::core::v1::ServiceAccount {
+                metadata: k8s::ObjectMeta {
+                    namespace: Some(self.namespace.clone()),
+                    name: Some("curl".to_string()),
+                    ..Default::default()
+                },
+                ..Default::default()
+            },
+        )
+        .await;
+
+        create(
+            &self.client,
+            k8s::api::rbac::v1::Role {
+                metadata: k8s::ObjectMeta {
+                    namespace: Some(self.namespace.clone()),
+                    name: Some("curl-lock".to_string()),
+                    ..Default::default()
+                },
+                rules: Some(vec![k8s::api::rbac::v1::PolicyRule {
+                    api_groups: Some(vec!["".to_string()]),
+                    resources: Some(vec!["configmaps".to_string()]),
+                    verbs: vec!["get".to_string(), "list".to_string(), "watch".to_string()],
+                    ..Default::default()
+                }]),
+            },
+        )
+        .await;
+
+        create(
+            &self.client,
+            k8s::api::rbac::v1::RoleBinding {
+                metadata: k8s::ObjectMeta {
+                    namespace: Some(self.namespace.clone()),
+                    name: Some("curl-lock".to_string()),
+                    ..Default::default()
+                },
+                role_ref: k8s::api::rbac::v1::RoleRef {
+                    api_group: "rbac.authorization.k8s.io".to_string(),
+                    kind: "Role".to_string(),
+                    name: "curl-lock".to_string(),
+                },
+                subjects: Some(vec![k8s::api::rbac::v1::Subject {
+                    kind: "ServiceAccount".to_string(),
+                    name: "curl".to_string(),
+                    namespace: Some(self.namespace.clone()),
+                    ..Default::default()
+                }]),
+            },
+        )
+        .await;
+    }
+
+    fn gen_pod(ns: &str, name: &str, target_url: &str, inject: LinkerdInject) -> k8s::Pod {
+        k8s::Pod {
+            metadata: k8s::ObjectMeta {
+                namespace: Some(ns.to_string()),
+                name: Some(name.to_string()),
+                annotations: Some(convert_args!(btreemap!(
+                    "linkerd.io/inject" => inject.to_string(),
+                    "config.linkerd.io/proxy-log-level" => "linkerd=trace,info",
+                ))),
+                ..Default::default()
+            },
+            spec: Some(k8s::PodSpec {
+                service_account: Some("curl".to_string()),
+                init_containers: Some(vec![k8s::api::core::v1::Container {
+                    name: "wait-for-nginx".to_string(),
+                    image: Some("docker.io/bitnami/kubectl:latest".to_string()),
+                    args: Some(
+                        vec![
+                            "wait",
+                            "--timeout=60s",
```

Comment:
If the watch fails the whole curl pod will fail, maybe I'm overthinking it a little bit but would 60s be enough for resources to get spun up?

Sometimes we see pods taking a longer time than normal to spin up in CI. In this case, we'd have to wait for server resources to be applied and sometimes for the nginx pod itself to properly spin up before we delete the lock. Not sure it's worth increasing the time but thought I'd ask regardless

Reply:
This reminds me that I didn't pull in the workflow changes to pre-pull the images... I'll do that.

I haven't seen the startup take all that long when the images are already on the cluster. We can always bump it in the future if we see failures.